src/osp/universe/universe.h

```diff
@@ -38,26 +38,96 @@
 namespace osp::universe
 {
 
-template <typename INT_T, std::size_t DIMENSIONS_T>
+/**
+ * @brief Describes strided vector data within an externally stored buffer
+ *
+ * Intended to make different data layouts easier to access, and allows
+ * interleving different datatypes:
+ * * { XXXXX ... YYYYY ... ZZZZZ ... }
+ * * { XYZ XYZ XYZ XYZ XYZ XYZ ... }
+ * * { XYZ ??? XYZ ??? XYZ ??? ... }
+ */
+template <typename T, int DIMENSIONS_T>
 struct SatVectorDesc
 {
-    using View_t = Corrade::Containers::StridedArrayView1D<INT_T>;
+    using View_t        = Corrade::Containers::StridedArrayView1D<T>;
+    using ViewConst_t   = Corrade::Containers::StridedArrayView1D<T const>;
+    using Data_t        = Corrade::Containers::ArrayView<unsigned char>;
+    using DataConst_t   = Corrade::Containers::ArrayView<unsigned char const>;
+
+    static constexpr int smc_dimensions = DIMENSIONS_T;
 
     // Offsets of xyz components in bytes
-    std::array<typename View_t::Size, DIMENSIONS_T> m_offsets;
+    std::array<std::size_t, DIMENSIONS_T> m_offsets;
 
     // Assuming xyz all have the same stride
-    typename View_t::Stride m_stride;
+    std::ptrdiff_t m_stride;
+
+    constexpr View_t view(Data_t data, std::size_t count, int dimension) const
+    {
+        using namespace Corrade::Containers;
```

Comment:
is this using namespace actually used?

Reply:
nope, likely had something else more complex here but removed it.